src/Raven.Server/Commercial/LetsEncrypt/SettingsZipFileHelper.cs

```diff
@@ -0,0 +1,582 @@
+                        {
+                            ms2.WriteByte((byte)'{');
+                            ms2.WriteByte((byte)'}');
+                            ms2.Position = 0;
+                            settingsJson = await context.ReadForMemoryAsync(ms2, "settings-json");
+                        }
+                    }
+        
+                    settingsJson.Modifications = new DynamicJsonValue(settingsJson);
+
+                    if (parameters.SetupMode == SetupMode.LetsEncrypt)
+                    {
+                        settingsJson.Modifications[RavenConfiguration.GetKey(x => x.Security.CertificateLetsEncryptEmail)] = parameters.SetupInfo.Email;
+
+                        try
+                        {
+                            var licenseString = JsonConvert.SerializeObject(parameters.SetupInfo.License, Formatting.Indented);
+
+                            var entry = archive.CreateEntry("license.json");
+                            entry.ExternalAttributes = ((int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR)) << 16;
+
+                            await using (var entryStream = entry.Open())
+                            await using (var writer = new StreamWriter(entryStream))
+                            {
+                                await writer.WriteAsync(licenseString);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            throw new InvalidOperationException("Failed to write license.json in zip archive.", e);
+                        }
+                    }
+
+                    settingsJson.Modifications[RavenConfiguration.GetKey(x => x.Core.SetupMode)] = parameters.SetupMode.ToString();
+                    
+                    if (parameters.SetupInfo.EnableExperimentalFeatures)
+                    {
+                        settingsJson.Modifications[RavenConfiguration.GetKey(x => x.Core.FeaturesAvailability)] = FeaturesAvailability.Experimental;
+                    }
+                    
+                    if (parameters.SetupInfo.Environment != StudioConfiguration.StudioEnvironment.None)
+                    {
+                        if (parameters.OnPutServerWideStudioConfigurationValues != null)
+                            await parameters.OnPutServerWideStudioConfigurationValues(parameters.SetupInfo.Environment);
+                    }
+                    
+                    var certificateFileName = $"cluster.server.certificate.{name}.pfx";
+                    string certPath;
+                    if (parameters.OnGetCertificatePath != null)
+                        certPath = await parameters.OnGetCertificatePath(certificateFileName);
+                    else
+                        certPath = Path.Combine(AppContext.BaseDirectory, certificateFileName);
+                    
+                    if (parameters.SetupInfo.ModifyLocalServer)
```

Comment:
This should be `false` when used from `rvn create-setup-package`. I see that `SetupInfo` is just parsed from file, so I guess that's the default. 

Could you add an assertion to tests that makes sure settings.json is *not* modified when running from `rvn` and is modified when run normally during the studio setup?

Reply:
done